crates/pallet-domains/src/lib.rs

```diff
@@ -584,6 +586,29 @@ impl<T: Config> Pallet<T> {
             }
         }
 
+        let bundle_created_on_valid_primary_block =
+            match pallet_receipts::PrimaryBlockHash::<T>::get(
+                DomainId::SYSTEM,
+                bundle.header.primary_number,
+            ) {
```

Comment:
Using the system domain's `PrimaryBlockHash` for the core domain may be inappropriate. For example, if a core domain is running slow or stalls and the corresponding `PrimaryBlockHash` of the system domain may have already been pruned, thus causing this check always fails for this core domain.

Reply:
Or perhaps we should skip further checking if `PrimaryBlockHash::<T>::get` return `None` and it is a core bundle, to make this check a best effort for the core domain.

Reply:
You're right, the check in pallet-domains is only suitable for the system domain bundle. But we need to find another way for the core domain bundles, skipping it for the core domain bundles does not solve the problem, we want it to be guaranteed, let me think more about this.

Reply:
Changed the corresponding check to system bundle only and added TODO for the core domains as it's not feasible right now.